pkg/sbproxy/notifications/handlers/broker_handler.go

```diff
@@ -107,12 +111,32 @@ func (bnh *BrokerResourceNotificationsHandler) OnCreate(ctx context.Context, pay
 		log.C(ctx).Debugf("Could not find platform broker in platform with name %s: %s", brokerToCreate.Resource.Name, err)
 	}
 
+	username, password, passwordHash, err := util.GenerateBrokerPlatformCredentials()
+	if err != nil {
+		log.C(ctx).Debugf("Could not generate broker platform credentials for broker (%s): %s", brokerToCreate.Resource.Name, err)
+		return
+	}
+
+	credentials := &types.BrokerPlatformCredential{
+		Username:       username,
+		PasswordHash:   passwordHash,
+		BrokerID:       brokerToCreate.Resource.ID,
+		NotificationID: notification.ID,
+	}
+
 	if existingBroker == nil {
 		log.C(ctx).Infof("Could not find platform broker in platform with name %s. Attempting to create a SM proxy registration...", brokerProxyName)
 
+		if err := bnh.SMClient.PutCredentials(ctx, credentials); err != nil {
```

Comment:
maybe we should first set the credentials in the platform and then set them in SM - this way we won't hit a case in which  we had multiple platform errors applying credentials update and sm having lost the current platform credentials